pychromecast/__init__.py

```diff
@@ -334,6 +364,68 @@ def join(self, timeout=None):
         """
         self.socket_client.join(timeout=timeout)
 
+    def start_dynamic_host_tracking(self):
+        """
+        Start dynamic host tracking, register for connection status events
+        """
+        self.socket_client.register_connection_listener(self)
+
+    def new_connection_status(self, status):
+        """
+        Callback for socket client connection status
+        """
+        if not status.status == CONNECTION_STATUS_DISCONNECTED:
+            return  # Not interesting
+
+        def discovery_timeout():
+            """
+            Internal scheduled timeout for discovery, 5 seconds
+            """
+            time.sleep(10)
+            if not self.browser:
+                return  # Change in ip detected, discovery did not time out
+
+            self.logger.info('Discovery timed out, stopping')
+            stop_discovery(self.browser)
+            self.browser = None
+            if not self.socket_client.is_connected:
+                # If no host on new ip address was found, try reconnecting
+                self.disconnect()
+                self.setup(self.host, self.port, self.device, self.kwargs)
+
+        self.logger.info('Dynamic host discovery started')
+        self.tracker, self.browser = \
+            start_discovery(self._dynamic_host_callback)
+
+        timeout = threading.Thread(target=discovery_timeout)
+        timeout.start()
+
+    def _dynamic_host_callback(self, unit_id):
+        """
+        Callback for dynamic discovery of host ip changes
+        """
+        unit = self.tracker.services[unit_id]
+        host = unit[0]
+        port = unit[1]
+        name = unit[4]
+
+        if not name == self.name or not port == self.port:
+            return  # Wrong group, keep discovering
+
+        if host == self.host:
+            return  # No change in host, keep discovering
+
+        self.logger.info('Change in host ip detected, '
+                         'reinitializing object...')
+        stop_discovery(self.browser)
+        self.browser = None
+        self.disconnect()
+        self.host = host
+        try:
+            self.setup(host, port, self.device, self.kwargs)
+        except ChromecastConnectionError:  # noqa
```

Comment:
To be fair, not sure. But flake8 throws an error otherwise.

I guess it has something to do with this [`noqa`](https://github.com/balloob/pychromecast/blob/a867ac9a2cbbdda7f1cd747259009b8dc073e3a2/pychromecast/__init__.py#L13)?